t/01-test_needle.t

```diff
@@ -63,6 +64,15 @@ $needle = needle->new($data_dir . "kde.ref.json");
 $res    = $img1->search($needle);
 ok(!defined $res, "no match with different art");
 
+try {
+    $img1 = tinycv::read($data_dir . "kde.ref.png");
+    $needle = needle->new($data_dir . "kde.ref.json");
+    my $needle_nopng = needle->new($data_dir . "console.ref.json");
+    $needle_nopng->{png} .= ".missing.png";
+    $res = $img1->search([$needle_nopng, $needle]);
+};
+ok(defined $res, "skip needles without png");
```

Comment:
I suggest to initialize the res - otherwise the next test added above relying on $res will make your test invalid.